### PR TITLE
Feature/product form single options

### LINKED ISSUE #1337: Products showing default "title" text when no variants exist in Shopify
### Description

When a course doesn't have variants we'd expect it to not display anything in the form, however it currently displays a default "title" and "default title" checkbox

### Link to Reproduction

https://osc-ecommerce.fly.dev/courses/a-level-biology

### Steps to reproduce

image

image

### Release Version

0.148.1

### Browser

_No response_

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Might be worth taking reference from the current site when these values aren't present e.g

https://www.openstudycollege.com/courses/anatomy-physiology-combined-3-4
image

https://www.openstudycollege.com/courses/access-2016-online-training
image

https://www.openstudycollege.com/courses/animal-behaviour-3
image

### PR DESCRIPTION
**Pull request by stevenbister**
- Closes #1337 

## 📝 Description

When courses don't have variants the default behaviour for Shopify is to return "title" and "default title" for product variants from the API. This means that if there are cases where we aren't adding variants to courses then we end up displaying the word "Title" and "Default Title" on the course page. E.g.

image

Similarly there might be cases where we are only providing one study method (e.g. online) and therefor seems unnecessary to display the form fields where they can't be interacted with.

I've updated the product form and the cart to remove the form fields where the default value is present and to display a message when only one study method is present. e.g.

image

image

This is purely frontend so the values will still pass through to the checkout if they are available within Shopify (a level biology in this case has no variants)

image

## ⛳️ Current behavior (updates)

The product form will display the study method and course options regardless of how many variants we have and will also display the default data for variants returned from the storefront API.

## 🚀 New behavior

The product form will no longer display the default data, and if only one study method variant is present it will display a message.

### CODE REVIEW

**Comment by mattwendzina** [comment]
Looks good mate. Couple of comments here:
Below is a screenshot of `/courses/anatomy-physiology-combined-3-4`. I'm pointing at Sanity Production and it says Study-Method rather than Study Options. I'm guessing this is just a case of where the variants haven't been updated for this product in Shopify?
Screenshot 2023-08-02 at 10 21 46

Just throwing this thought out there... wonder if it's worth doing the same for Course Options where there isn't any options. This way it would be clear what the course option is and would keep a consistent look across all of the products.

**Review by mattwendzina — CHANGES_REQUESTED** [review]
Hey mate, looking good. Left a couple of comments.

File: packages/osc-ecommerce/app/utils/storefront.helpers.spec.ts [hunk 1]
```diff
@@ -77,3 +78,15 @@ describe('isAddOn', () => {
         expect(isAddOn(productVariant)).toBe(true);
     });
 });
+
+describe('hasStudyOption', () => {
+    test('returns true if option name matches', () => {
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-ecommerce/app/utils/storefront.helpers.spec.ts:83]: Aha yeah, we can probably ditch that & `__DEV__ProductOptions`. I'll probs do that outside of this PR though

**Comment by stevenbister** [comment]
> Looks good mate. Couple of comments here: Below is a screenshot of `/courses/anatomy-physiology-combined-3-4`. I'm pointing at Sanity Production and it says Study-Method rather than Study Options. I'm guessing this is just a case of where the variants haven't been updated for this product in Shopify? Screenshot 2023-08-02 at 10 21 46

Looks okay on my end? You defo got the prod creds?
image

> Just throwing this thought out there... wonder if it's worth doing the same for Course Options where there isn't any options. This way it would be clear what the course option is and would keep a consistent look across all of the products.

Looks like Marketing are asking to leave it off

I also noticed an eslint error so have pushed up a little fix for that

**Comment by mattwendzina** [comment]
> > Looks good mate. Couple of comments here: Below is a screenshot of `/courses/anatomy-physiology-combined-3-4`. I'm pointing at Sanity Production and it says Study-Method rather than Study Options. I'm guessing this is just a case of where the variants haven't been updated for this product in Shopify? Screenshot 2023-08-02 at 10 21 46
> 
> Looks okay on my end? You defo got the prod creds? image
> 
> > Just throwing this thought out there... wonder if it's worth doing the same for Course Options where there isn't any options. This way it would be clear what the course option is and would keep a consistent look across all of the products.
> 
> Looks like Marketing are asking to leave it off
> 
> I also noticed an eslint error so have pushed up a little fix for that

Cool. All sounds good. I had another check and it does seem to be displaying correctly now 👍

**Review by mattwendzina — APPROVED** [review]
Looks good mate 🚢 
Prob just worth getting a ticket for updating tests/removing the `__DEV__ProductOptions` at some point so it doesn't get forgotten 👍

**Pull request merged**